engine/table/src/main/java/io/deephaven/engine/table/impl/select/DynamicWhereFilter.java

```diff
@@ -70,46 +71,77 @@ public DynamicWhereFilter(final QueryTable setTable, final boolean inclusion, fi
         if (setRefreshing) {
             this.setTable = setTable;
             setTupleSource = TupleSourceFactory.makeTupleSource(setColumns);
-            setTable.getRowSet().forAllRowKeys((final long v) -> addKey(makeKey(v)));
+            try (final CloseableIterator<?> initialKeysIterator = ChunkedColumnIterator.make(
+                    setTupleSource, setTable.getRowSet())) {
+                initialKeysIterator.forEachRemaining(this::addKey);
+            }
 
-            final String[] columnNames = Arrays.stream(matchPairs).map(MatchPair::rightColumn).toArray(String[]::new);
-            final ModifiedColumnSet modTokenSet = setTable.newModifiedColumnSet(columnNames);
+            final String[] setColumnNames =
+                    Arrays.stream(matchPairs).map(MatchPair::rightColumn).toArray(String[]::new);
+            final ModifiedColumnSet setColumnsMCS = setTable.newModifiedColumnSet(setColumnNames);
             setUpdateListener = new InstrumentedTableUpdateListenerAdapter(
                     "DynamicWhereFilter(" + Arrays.toString(setColumnsNames) + ")", setTable, false) {
 
                 @Override
                 public void onUpdate(final TableUpdate upstream) {
-                    if (upstream.added().isEmpty() && upstream.removed().isEmpty()
-                            && !upstream.modifiedColumnSet().containsAny(modTokenSet)) {
+                    final boolean hasAdds = upstream.added().isNonempty();
+                    final boolean hasRemoves = upstream.removed().isNonempty();
+                    final boolean hasModifies = upstream.modified().isNonempty()
+                            && upstream.modifiedColumnSet().containsAny(setColumnsMCS);
+                    if (!hasAdds && !hasRemoves && !hasModifies) {
                         return;
                     }
 
-                    final MutableBoolean trueModification = new MutableBoolean(false);
+                    // Remove removed keys
+                    if (hasRemoves) {
+                        try (final CloseableIterator<?> removedKeysIterator = ChunkedColumnIterator.make(
+                                setTupleSource.getPrevSource(), upstream.removed())) {
+                            removedKeysIterator.forEachRemaining(DynamicWhereFilter.this::removeKey);
+                        }
+                    }
 
-                    upstream.added().forAllRowKeys((final long v) -> addKey(makeKey(v)));
-                    upstream.removed().forAllRowKeys((final long v) -> removeKey(makePrevKey(v)));
```

Comment:
I see the order changed from added, removed, modified to added, modified, removed. Just double checking, that's intentional (looks OK to do based on code).

Reply:
It's actually removes, then mods, then adds. These orders are equally correct. Note that setTable always has distinct keys.

I changed it for two reasons:
1. This is a more canonical order.
2. This order keeps the hash table at its minimum possible size, which may avoid unnecessary expansions/rehashes/collisions that will be mooted by subsequent operations.